boards/nordic/nrf54l15pdk/nrf54l15pdk_nrf54l15-common.dtsi

```diff
@@ -98,3 +98,10 @@
 	pinctrl-1 = <&pwm20_sleep>;
 	pinctrl-names = "default", "sleep";
 };
+
+arduino_i2c: &i2c21 {
```

Comment:
I don't entirely understand this change. There is no Arduino header on this board. Are the shields supposed to be connected with some extra wiring? Or am I missing something?

Reply:
actually, good point… I assumed board followed the good old layouts.

Reply:
Yes, there is extra wiring needed (e.g.: https://github.com/nrfconnect/sdk-nrf/pull/16746/files#diff-2acb4979b946281593b13ff0b639cc04873f374cf9369961ba8c2028b01e5ee5)

Shields that use I2C rely on `arduino_i2c` being defined, which has to be done in the board overlay.
Unfortunately it's not possible to specify it in the application overlay due to the order the overlays are applied.

Reply:
So then, an arduino header can't be added.

Reply:
> Unfortunately it's not possible to specify it in the application overlay due to the order the overlays are applied.

I see. And I think such hack is acceptable since this is a temporary board. I'd just add a comment to clarify the situation so there's no confusion regarding the Arduino header.

Reply:
This could be handled downstream by skipping the shield for this board and putting the full configuration in the overlay instead.
@gmarull would this be your preferred approach?

Reply:
you could still asign pins here, but skip the `arduino_i2c` label, you can then have a board-specific overlay in a shield I guess?

Reply:
Downstream implementation:
https://github.com/nrfconnect/sdk-nrf/pull/16746/files